### ISSUE TITLE
[App]: Ray Tune is not working with WandB

### ISSUE THREAD

**Issue by zshareef** [wandb/wandb#6038]
### Current Behavior

Currently I am trying Ray Tune with WandB. I am trying the following example given on the following page:
https://docs.ray.io/en/latest/tune/examples/tune-wandb.html#setup-wandb

I am trying it using `setup_wandb` utility. The complete code is given below:

```
import os
import ray
import numpy as np
from ray import tune
from ray.air import session
from ray.air.integrations.wandb import setup_wandb

def train_function_wandb(config):
    wandb = setup_wandb(api_key="my-API-Key",
                        project=config.get("wandb_project_2"))

    for i in range(30):
        loss = config["mean"] + config["sd"] * np.random.randn()
        session.report({"loss": loss})
        wandb.log({"loss": loss})


def tune_with_setup():
    """Example for using the setup_wandb utility with the function API"""
    tuner = tune.Tuner(
        train_function_wandb,
        tune_config=tune.TuneConfig(
            metric="loss",
            mode="min",
        ),
        param_space={
            "mean": tune.grid_search([1, 2, 3, 4, 5]),
            "sd": tune.uniform(0.2, 0.8),
        },
    )
    tuner.fit()


mock_api = False

if mock_api:
    os.environ.setdefault("WANDB_MODE", "disabled")
    os.environ.setdefault("WANDB_API_KEY", "my-API-Key")
    os.environ.setdefault("WANDB_BASE_URL", "http://smtcig00.wandb.127.0.0.1.sslip.io:8000/")
    ray.init(
        runtime_env={"env_vars": {"WANDB_MODE": "disabled",
                                  "WANDB_API_KEY": "my-API-Key",
                                  "WANDB_BASE_URL": "http://smtcig00.wandb.127.0.0.1.sslip.io:8000/"}}
    )

tune_with_setup()
```

### Expected Behavior

I should get the loss function on WandB.

### Screenshots

I am getting the following Error. I am not able to connect to WandB server. The following is the log.

```
(train_function_wandb pid=48305) wandb: Network error (ConnectionError), entering retry loop.
(train_function_wandb pid=48305) wandb: W&B API key is configured. Use `wandb login --relogin` to force relogin
(train_function_wandb pid=48305) wandb: - Waiting for wandb.init()...
(train_function_wandb pid=48305) wandb: \ Waiting for wandb.init()...
(train_function_wandb pid=48305) wandb: | Waiting for wandb.init()...
(train_function_wandb pid=48305) wandb: Network error (ConnectionError), entering retry loop. [repeated 5x across cluster] (Ray deduplicates logs by default. Set RAY_DEDUP_LOGS=0 to disable log deduplication, or see https://docs.ray.io/en/master/ray-observability/ray-logging.html#log-deduplication for more options.)
(train_function_wandb pid=48305) wandb: / Waiting for wandb.init()...
(train_function_wandb pid=48306) wandb: W&B API key is configured. Use `wandb login --relogin` to force relogin [repeated 4x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48309) wandb: | Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48307) wandb: Network error (ConnectionError), entering retry loop. [repeated 4x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48309) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48305) wandb: - Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: \ Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: | Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: / Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48309) wandb: - Waiting for wandb.init()... [repeated 9x across cluster]
Trial status: 5 RUNNING
Current time: 2023-08-10 10:02:39. Total running time: 30s
Logical resource usage: 5.0/10 CPUs, 0/0 GPUs
╭─────────────────────────────────────────────────────────────────╮
│ Trial name                         status           sd     mean │
├─────────────────────────────────────────────────────────────────┤
│ train_function_wandb_3425b_00000   RUNNING    0.754165        1 │
│ train_function_wandb_3425b_00001   RUNNING    0.293812        2 │
│ train_function_wandb_3425b_00002   RUNNING    0.654419        3 │
│ train_function_wandb_3425b_00003   RUNNING    0.766079        4 │
│ train_function_wandb_3425b_00004   RUNNING    0.445314        5 │
╰─────────────────────────────────────────────────────────────────╯

(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48309) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48305) wandb: \ Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: | Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48305) wandb: - Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48305) wandb: / Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
Trial status: 5 RUNNING
Current time: 2023-08-10 10:03:09. Total running time: 1min 0s
Logical resource usage: 5.0/10 CPUs, 0/0 GPUs
```

### Environment

OS: MacBook Pro M1

Browsers: Chrome

Version:

### Additional Context

I am currently running this example on MacBook Pro M1.
I am not using the WandB setup (wandb.ai). We have setup our own infrastructure to deploy WandB at the address:
http://smtcig00.wandb.127.0.0.1.sslip.io:8000/
We are accessing this web address using VPN.

Meanwhile I have tried another example of Ray with WandB given on the Ray website. In this example I am also using `setup_wandb` API.
https://docs.ray.io/en/latest/ray-air/examples/upload_to_wandb.html
But here I am able to log the data on the WandB. Here I am using the same setup as described above mean using VPN and our own setup.

**Comment by sydholl** [comment]
Internal GitHub User commented: 
### Current Behavior

Currently I am trying Ray Tune with WandB. I am trying the following example given on the following page:
https://docs.ray.io/en/latest/tune/examples/tune-wandb.html#setup-wandb

I am trying it using `setup_wandb` utility. The complete code is given below:

```
import os
import ray
import numpy as np
from ray import tune
from ray.air import session
from ray.air.integrations.wandb import setup_wandb

def train_function_wandb(config):
    wandb = setup_wandb(api_key="my-API-Key",
                        project=config.get("wandb_project_2"))

    for i in range(30):
        loss = config["mean"] + config["sd"] * np.random.randn()
        session.report({"loss": loss})
        wandb.log({"loss": loss})


def tune_with_setup():
    """Example for using the setup_wandb utility with the function API"""
    tuner = tune.Tuner(
        train_function_wandb,
        tune_config=tune.TuneConfig(
            metric="loss",
            mode="min",
        ),
        param_space={
            "mean": tune.grid_search([1, 2, 3, 4, 5]),
            "sd": tune.uniform(0.2, 0.8),
        },
    )
    tuner.fit()


mock_api = False

if mock_api:
    os.environ.setdefault("WANDB_MODE", "disabled")
    os.environ.setdefault("WANDB_API_KEY", "my-API-Key")
    os.environ.setdefault("WANDB_BASE_URL", "http://smtcig00.wandb.127.0.0.1.sslip.io:8000/")
    ray.init(
        runtime_env={"env_vars": {"WANDB_MODE": "disabled",
                                  "WANDB_API_KEY": "my-API-Key",
                                  "WANDB_BASE_URL": "http://smtcig00.wandb.127.0.0.1.sslip.io:8000/"}}
    )

tune_with_setup()
```

### Expected Behavior

I should get the loss function on WandB.

### Steps To Reproduce

_No response_

### Screenshots

I am getting the following Error. I am not able to connect to WandB server. The following is the log.

```
(train_function_wandb pid=48305) wandb: Network error (ConnectionError), entering retry loop.
(train_function_wandb pid=48305) wandb: W&B API key is configured. Use `wandb login --relogin` to force relogin
(train_function_wandb pid=48305) wandb: - Waiting for wandb.init()...
(train_function_wandb pid=48305) wandb: \ Waiting for wandb.init()...
(train_function_wandb pid=48305) wandb: | Waiting for wandb.init()...
(train_function_wandb pid=48305) wandb: Network error (ConnectionError), entering retry loop. [repeated 5x across cluster] (Ray deduplicates logs by default. Set RAY_DEDUP_LOGS=0 to disable log deduplication, or see https://docs.ray.io/en/master/ray-observability/ray-logging.html#log-deduplication for more options.)
(train_function_wandb pid=48305) wandb: / Waiting for wandb.init()...
(train_function_wandb pid=48306) wandb: W&B API key is configured. Use `wandb login --relogin` to force relogin [repeated 4x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48309) wandb: | Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48307) wandb: Network error (ConnectionError), entering retry loop. [repeated 4x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48309) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48305) wandb: - Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: \ Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: | Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: / Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48309) wandb: - Waiting for wandb.init()... [repeated 9x across cluster]
Trial status: 5 RUNNING
Current time: 2023-08-10 10:02:39. Total running time: 30s
Logical resource usage: 5.0/10 CPUs, 0/0 GPUs
╭─────────────────────────────────────────────────────────────────╮
│ Trial name                         status           sd     mean │
├─────────────────────────────────────────────────────────────────┤
│ train_function_wandb_3425b_00000   RUNNING    0.754165        1 │
│ train_function_wandb_3425b_00001   RUNNING    0.293812        2 │
│ train_function_wandb_3425b_00002   RUNNING    0.654419        3 │
│ train_function_wandb_3425b_00003   RUNNING    0.766079        4 │
│ train_function_wandb_3425b_00004   RUNNING    0.445314        5 │
╰─────────────────────────────────────────────────────────────────╯

(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48309) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48305) wandb: \ Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48305) wandb: | Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48305) wandb: - Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48305) wandb: / Waiting for wandb.init()... [repeated 6x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 9x across cluster]
(train_function_wandb pid=48306) wandb: - Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: \ Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: | Waiting for wandb.init()... [repeated 5x across cluster]
(train_function_wandb pid=48306) wandb: / Waiting for wandb.init()... [repeated 5x across cluster]
Trial status: 5 RUNNING
Current time: 2023-08-10 10:03:09. Total running time: 1min 0s
Logical resource usage: 5.0/10 CPUs, 0/0 GPUs
```

### Environment

OS: MacBook Pro M1

Browsers: Chrome

Version:


### Additional Context

I am currently running this example on MacBook Pro M1.
I am not using the WandB setup (wandb.ai). We have setup our own infrastructure to deploy WandB at the address:
http://smtcig00.wandb.127.0.0.1.sslip.io:8000/
We are accessing this web address using VPN.

Meanwhile I have tried another example of Ray with WandB given on the Ray website. In this example I am also using `setup_wandb` API.
https://docs.ray.io/en/latest/ray-air/examples/upload_to_wandb.html
But here I am able to log the data on the WandB. Here I am using the same setup as described above mean using VPN and our own setup.

**Comment by nate-wandb** [comment]
Hi @zshareef, it looks like the logic in your code would ignore the if block and attempt to connect to wandb.ai. Is this what you are expecting or maybe I'm misunderstanding your goal here.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
Auto-assigned

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by ChenchenHu007** [comment]
when I initialized the wandb by setup_wandb, wandb.log() does not log anything. I cannot find any custom metrics in wandb.ai website.

**Comment by sydholl** [comment]
We are closing this issue due to inactivity. If you feel that it merits additional review, please provide a reproduction and add a comment explaining why. Thank you.
